explorer/client/src/components/ownedobjects/OwnedObjects.tsx

```diff
@@ -33,174 +32,162 @@ const DATATYPE_DEFAULT: resultType = [
     },
 ];
 
-function OwnedObjectStatic({ objects }: { objects: string[] }) {
-    const results = objects.map((objectId) => {
-        const entry = findDataFromID(objectId, undefined);
-        return {
-            id: entry?.id,
-            Type: entry?.objType,
-            Version: entry?.version,
-            display: entry?.data?.contents?.display,
-        };
-    });
-
-    return <OwnedObjectView results={results} />;
+const IS_COIN_TYPE = (typeDesc: string): boolean => /::Coin::/.test(typeDesc);
+
+function OwnedObject({ id }: { id: string }) {
+    if (process.env.REACT_APP_DATA === 'static') {
+        return <OwnedObjectStatic id={id} />;
+    } else {
+        return <OwnedObjectAPI id={id} />;
+    }
 }
 
-function OwnedObjectAPI({ objects }: { objects: string[] }) {
+function OwnedObjectStatic({ id }: { id: string }) {
+    const objects = findOwnedObjectsfromID(id);
+
+    if (objects) {
+        const results = objects?.map(({ objectId }) => {
+            const entry = findDataFromID(objectId, undefined);
+            return {
+                id: entry?.id,
+                Type: entry?.objType,
+                Version: entry?.version,
+                display: entry?.data?.contents?.display,
+                balance: entry?.data?.contents?.balance,
+            };
+        });
+
+        return <GroupView results={results} />;
+    } else {
+        return <div />;
+    }
+}
+
+function OwnedObjectAPI({ id }: { id: string }) {
     const [results, setResults] = useState(DATATYPE_DEFAULT);
     const [isLoaded, setIsLoaded] = useState(false);
 
+    // TODO - The below will fail for a large number of owned objects
+    // due to the number of calls to the API
+    // Backend changes will be required to enable a scalable solution
+    // getOwnedObjectRefs will need to return id, type and balance for each owned object
```

Comment:
we may not want to make getOwnedObjectRefs always return this data, as it adds a lot. likely, we should either:
* add another, similar route with more info
* parameterize the route, to optionally include the information

Reply:
I agree - there's definitely a balancing act to be struck here. @666lcz --> is there a GitHub issue on extending the info provided by `getOwnedObjectRefs` so we can funnel discussion here?